### PR TITLE
Irnmn-slider component and readme file

### PR DESCRIPTION
**Pull request by samuelsilvapt**
Simple web component that creates a slideshow - I'm testing it in this ticket: https://ennismore.atlassian.net/browse/IRNMN-1975 , attaching the component to a WP block.

### CODE REVIEW

File: slider/index.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+        if (this.dataset.sliderInitialized === "true") {
+            return;
+        }
+        this.dataset.sliderInitialized = "true";
+
+        /**
+         * Handle slider loopping
+         */
+        const firstClone = slides[0].cloneNode(true);
```
Comment (ejcromwell) [inline slider/index.js:51]: @samuelsilvapt Are you handling  a scenario where only one image is provided? Would this logic still make sense?

File: slider/index.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        this.querySelector(this.CLASSNAMES.TOTAL_SLIDES).textContent = totalSlides;
+
+
+        /**
+         * Update slides position using translateX method
+         * 
+         * @returns { void }
+         */
+        const updateSlides = () => {
+            swipeContainer.style.transition = 'transform 0.3s ease';
```
Comment (ejcromwell) [inline slider/index.js:69]: @samuelsilvapt is there a way we can move the ease and timing values to a CSS custom property so they can be easily overwritten?

File: slider/index.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    /**
+     * Get the selectors from the attribute
+     * to be used as classnames obj
+     * 
+     * @returns {object} - The selectors object
+     */
+    get selectors() {
+        let selectors = this.getAttribute('selectors');
```
Comment (ejcromwell) [inline slider/index.js:18]: @samuelsilvapt Feels like a try/ catch could offer a bit of error handling in case no json or correct format.

**Review by ejcromwell — CHANGES_REQUESTED** [review]
Overall it feels like it needs a bit more error handing or handling of edge cases not slides, one slide etc... 

The `initSlider()` seems too big for no real reason I can see. Would be easier to read and extend if functionality like events listners, silde output, next/ prev logic, touch events were all moved to there own functions. Example would be if I want to show 3 slides per view instead of one in the future. Would be easier if slide logic was seperate so we could call function depending on argument passed to component.

**Comment by samuelsilvapt** [comment]
@ejcromwell you can now find a better approach to the initSlider and some improvements based on your feedback - I'm commenting them individually

File: slider/index.js (hunk 1 above)
Comment (samuelsilvapt) [reply slider/index.js:105]: I added a new method to handle the one slider only fallback - following swiper logic, renderSingleSlide removes the navigation if exists and display the single slide normally

File: slider/index.js (hunk 2 above)
Comment (samuelsilvapt) [reply slider/index.js]: I added an option to the transition be defined in the HTML props

File: slider/index.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            else updateSlides(); // Reset to the current slide if swipe distance is insufficient
+            currentX = 0;
+            isDragging = false;
+        };
+
+        /**
+         * Add event listeners to the swipe wrapper
+         * 
+         */
+        swipeContainer.addEventListener('touchstart', touchStart);
```
[the remark this answers is not in the document]
Comment (samuelsilvapt) [reply slider/index.js]: I created some separated methods to handle the listeners (adding and removing them). You'll see an object with all the listeners

File: slider/index.js (hunk 3 above)
Comment (samuelsilvapt) [reply slider/index.js:18]: I added try/catch in everything I found necessary. This case, also number of sliders

File: slider/index.js [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+    }
+
+    /**
+     * Update slides position and pagination
+     */
+    updateSlides(swipeContainer, clonedSlidesCount, totalSlides) {
+        swipeContainer.style.transition = this.transition;
+        swipeContainer.style.transform = `translateX(-${this.currentSlide * 100}%)`;
+
+        const displayedSlideIndex =
```
Comment (ejcromwell) [inline slider/index.js:190]: @samuelsilvapt it's quite difficult to read nested ternary statements. To me making it a bit more verbose really helps to be able to easily follow the logic. Something like:

```
let displayedSlideIndex;

if (this.currentSlide === 0) {
    displayedSlideIndex = totalSlides;
    return displayedSlideIndex;
}

if (this.currentSlide === clonedSlidesCount - 1) {
    displayedSlideIndex = 1;
    return displayedSlideIndex;
}

displayedSlideIndex = this.currentSlide;
return displayedSlideIndex;
```

Not as sexy, but easier for me to process.

Comment (samuelsilvapt) [reply slider/index.js]: @ejcromwell I've replaced this with a switch case

**Pull request merged**